src/main/java/org/testng/TestRunner.java

```diff
@@ -1056,7 +1056,20 @@ public XmlTest getTest() {
 
   @Override
   public List<IConfigurationListener> getConfigurationListeners() {
-    return Lists.newArrayList(m_configurationListeners);
+    List<IConfigurationListener> listeners = Lists.newArrayList(m_configurationListeners);
+    for (IConfigurationListener each : this.m_configuration.getConfigurationListeners()) {
+      boolean duplicate = false;
+      for (IConfigurationListener listener : listeners) {
```

Comment:
If you use a Map, you'll be able to remove this loop

Reply:
Using a `Map` will make this more complex. We basically want all items in `m_configurationListeners` but we want only those items from `m_configuration.getConfigurationListeners()` which aren't already present in our list. I felt the current logic was a bit more easier to manage. Using a map(I think we would need a multi map) would later require me to additionally flatten the values as well into a single collection (because it would be a collection of collections initially)